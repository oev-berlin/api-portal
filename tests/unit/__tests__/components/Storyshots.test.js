import initStoryshots from '@storybook/addon-storyshots';
import { imageSnapshot } from '@storybook/addon-storyshots-puppeteer';
import playwright from 'playwright';

initStoryshots({
  suite: 'DOM snapshots',
});

initStoryshots({
  suite: 'Image snapshots',
  test: imageSnapshot({
    testTimout: 30_0000,
    getCustomBrowser: async () => {
      const browser = await playwright.chromium.launch();
      const context = await browser.newContext({
        viewport: { width: 800, height: 600 },
      });
      return context;
    },
  }),
});
