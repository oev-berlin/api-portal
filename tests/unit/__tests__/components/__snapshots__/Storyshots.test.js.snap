// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM snapshots Button Contained Primary Button 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="submit"
>
  Button
</button>
`;

exports[`DOM snapshots Button Outlined Secondary Button 1`] = `
<button
  className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="submit"
>
  Button
</button>
`;

exports[`DOM snapshots Button Text Button 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root  css-kpm0tx-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="submit"
>
  Button
</button>
`;

exports[`DOM snapshots Card Card Complete 1`] = `
<div
  className="sc-gsDKAQ kleci"
>
  <h3
    className="sc-dkPtRN ldyoeA"
  >
    Hello World
  </h3>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27200%27%20height=%27100%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="Tier"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="Tier"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fimage.geo.de%2F30147338%2Ft%2FiL%2Fv4%2Fw1440%2Fr0%2F-%2Fnilpferd-wildlife-photographer-of-the-year-jpg--83937-.jpg&w=640&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fimage.geo.de%2F30147338%2Ft%2FiL%2Fv4%2Fw1440%2Fr0%2F-%2Fnilpferd-wildlife-photographer-of-the-year-jpg--83937-.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.geo.de%2F30147338%2Ft%2FiL%2Fv4%2Fw1440%2Fr0%2F-%2Fnilpferd-wildlife-photographer-of-the-year-jpg--83937-.jpg&w=640&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
</div>
`;

exports[`DOM snapshots Card Card Image 1`] = `
<div
  className="sc-gsDKAQ kleci"
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27200%27%20height=%27100%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt=""
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt=""
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fimage.geo.de%2F30147338%2Ft%2FiL%2Fv4%2Fw1440%2Fr0%2F-%2Fnilpferd-wildlife-photographer-of-the-year-jpg--83937-.jpg&w=640&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fimage.geo.de%2F30147338%2Ft%2FiL%2Fv4%2Fw1440%2Fr0%2F-%2Fnilpferd-wildlife-photographer-of-the-year-jpg--83937-.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.geo.de%2F30147338%2Ft%2FiL%2Fv4%2Fw1440%2Fr0%2F-%2Fnilpferd-wildlife-photographer-of-the-year-jpg--83937-.jpg&w=640&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
</div>
`;

exports[`DOM snapshots Card Card Title 1`] = `
<div
  className="sc-gsDKAQ kleci"
>
  <h3
    className="sc-dkPtRN ldyoeA"
  >
    Hello World
  </h3>
</div>
`;
